solarforecastarbiter/io/reference_observations/eia.py

```diff
@@ -0,0 +1,141 @@
+    """Creates an observation at the site.
+
+    Parameters
+    ----------
+    api : solarforecastarbiter.io.api.APISession
+        API Session object, authenticated for the Reference user.
+    site : solarforecastarbiter.datamodel.Site
+        The site object for which to create the Observations.
+
+    Notes
+    -----
+    Currently only creates observations for net load [MW]
+    (`f"EBA.{eia_site_id}.D.H"`), but EIA contains other variables that may be
+    incorporated later (e.g. solar generation:
+    `f"EBA.{eia_site_id}.NG.SUN.H"`).
+
+    """
+
+    sfa_var = "net_load"
+    logger.info(f'Creating {sfa_var} at {site.name}')
+    try:
+        common.create_observation(api, site, sfa_var)
+    except HTTPError as e:
+        logger.error(f'Could not create Observation for "{sfa_var}" '
+                     f'at EIA site {site.name}')
+        logger.debug(f'Error: {e.response.text}')
+
+
+def initialize_site_forecasts(api, site):
+    """Creates a forecast at the site.
+
+    Parameters
+    ----------
+    api : solarforecastarbiter.io.api.APISession
+        API Session object, authenticated for the Reference user.
+    site : solarforecastarbiter.datamodel.Site
+        The site object for which to create the Observations.
+
+    """
+    common.create_forecasts(
+        api, site, ["net_load"],
+        default_forecasts.TEMPLATE_NETLOAD_PERSISTENCE_FORECASTS)
+
+
+def fetch(api, site, start, end, *, eia_api_key):
+    """Retrieve observation data for a EIA site between start and end.
+
+    Parameters
+    ----------
+    api : solarforecastarbiter.io.APISession
+        Unused but conforms to common.update_site_observations call
+    site : solarforecastarbiter.datamodel.Site
+        Site object with the appropriate metadata.
+    start : datetime
+        The beginning of the period to request data for.
+    end : datetime
+        The end of the period to request data for.
+    eia_api_key : str
+        API key for api.eia.gov
+
+    Returns
+    -------
+    data : pandas.DataFrame
+        All of the requested data as a single DataFrame.
+
+    Notes
+    -----
+    Currently only fetches observations for net load [MW]
+    (`f"EBA.{eia_site_id}.D.H"`), but EIA contains other variables that may be
+    incorporated later (e.g. solar generation:
+    `f"EBA.{eia_site_id}.NG.SUN.H"`).
+
+    """
+    try:
+        site_extra_params = common.decode_extra_parameters(site)
+    except ValueError:
+        return pd.DataFrame()
+    eia_site_id = site_extra_params['network_api_id']
+    series_id = f"EBA.{eia_site_id}.D.H"   # hourly net load (demand)
+    obs_df = eia.get_eia_data(
+        series_id, eia_api_key,
+        start,
+        end
+    )
+    if obs_df.empty:
+        logger.warning(f'Data for site {site.name} contained no '
+                       f'entries from {start} to {end}.')
+        return pd.DataFrame()
+    obs_df = obs_df.rename(columns={"value": "net_load"})
```

Comment:
The PVDAQ API returns times in the tz of the site, but they are not explicitly localized. For example, for noon at a site within US Mountain time zone, you get a timestamp like `2019-06-25 12:00` rather than `2019-06-25 12:00-0700`. The `tz_localize(site.timezone)` addresses this. We could have added a `.tz_convert('UTC')` after that but it doesn't really matter since the API performs the conversion.

So long as `io.fetch.eia.get_eia_data()` returns data localized to something then we should be ok.